core/src/main/java/net/opentsdb/query/processor/downsample/DownsampleNumericToNumericArrayIterator.java

```diff
@@ -0,0 +1,577 @@
+      TimeSeriesValue<NumericType> next = (TimeSeriesValue<NumericType>) iterator.next();
+
+      if (next.timestamp().compare(Op.GTE, this.interval_start)
+          && next.timestamp().compare(Op.LT, this.interval_end)) {
+
+        if (nextTs == null) {
+          // Use the first valid dp timestamp as the first Timestamp.
+          nextTs = next.timestamp().getCopy();
+          long nextRoundoffTs =
+              next.timestamp().epoch() - (next.timestamp().epoch() % intervalPart) + intervalPart;
+          nextTs.updateEpoch(nextRoundoffTs);
+        }
+
+        if (next.timestamp().compare(Op.GTE, nextTs)) {
+          // flush previous interval once new interval starts
+          flush = true;
+
+          // find the index to update in the Aggregator
+          index = (int) (((nextTs.epoch() - interval_start.epoch() - intervalPart) / intervalPart)
+              % intervals);
+
+          // get the next timestamp in the downsample
+          long nextTsRoundoff =
+              next.timestamp().epoch() - (next.timestamp().epoch() % intervalPart) + intervalPart;
+          nextTs.updateEpoch(nextTsRoundoff);
+        }
+        
+        if (accumulator.isFull() || flush) {
+          accumulateDoubles(accumulator, localLongAggs, localDoubleAggs);
+          accumulateLongs(accumulator, localLongAggs, localDoubleAggs);
+
+          accumulator.reset();
+        }
+        if (next.value().isInteger()) {
+          accumulator.add(next.value().longValue());
+        } else {
+          accumulator.add(next.value().doubleValue());
+        }
+
+        if (flush) {
+
+          // increment interval in the groupby aggregator
+          double v = getAggValue(localDoubleAggs, localLongAggs);
+
+          if (numericArrayAggregator instanceof ArrayCount && this.aggEnum == AggEnum.count) {
+            ((ArrayCount) numericArrayAggregator).accumulate(v, index, true);
+          } else {
+            numericArrayAggregator.accumulate(v, index);
+          }
+          // reset Local aggregates
+          resetLocalAggs(localDoubleAggs, localLongAggs);
+          flush = false;
+        }
+
+      }
+    }
+    if (nextTs != null) {
+      // capture the last entry in agg interval
+      accumulateDoubles(accumulator, localLongAggs, localDoubleAggs);
+      accumulateLongs(accumulator, localLongAggs, localDoubleAggs);
+      index = (int) (((nextTs.epoch() - interval_start.epoch() - intervalPart) / intervalPart)
+          % intervals);
+      double v = getAggValue(localDoubleAggs, localLongAggs);
+      if (numericArrayAggregator instanceof ArrayCount && this.aggEnum == AggEnum.count) {
+        ((ArrayCount) numericArrayAggregator).accumulate(v, index, true);
+      } else {
+        numericArrayAggregator.accumulate(v, index);
+      }
+    }
+
+    return null;
+  }
+
+  private double getAggValue(double[] localDoubleAggs, long[] localLongAggs) {
+    double v = Double.NaN;
+    switch (this.aggEnum) {
```

Comment:
This is where should still be able to use the new agg interface for array aggregators.